nixos/modules/services/misc/matrix-conduit.nix

```diff
@@ -0,0 +1,220 @@
+            };
+            global.allow_jaeger = mkOption {
+              type = types.bool;
+              default = false;
+              description = "Whether support monitoring and troubleshooting through jaeger is enabled on this server.";
+            };
+            global.trusted_servers = mkOption {
+              type = types.listOf types.str;
+              default = [ "matrix.org" ];
+              description = "Servers trusted with signing server keys.";
+            };
+
+            global.address = mkOption {
+              type = types.str;
+              default = "::1";
+              description = "Address to listen on for connections by the reverse proxy/tls terminator.";
+            };
+            global.database_path = mkOption {
+              type = types.str;
+              default = "/var/lib/matrix-conduit/conduit.db";
+              readOnly = true;
+              description = ''
+                Path to the conduit database, the directory where conduit will save its data.
+                Note that due to using the DynamicUser feature of systemd, this value should not be changed
+                and is set to be read only.
+              '';
+            };
+          };
+        };
+        default = {};
+        description = ''
+            Generates the conduit.toml configuration file. Refer to
+            <link xlink:href="https://gitlab.com/famedly/conduit/-/blob/master/conduit-example.toml"/>
+            for details on supported values.
+            Note that database_path should not be set because the services reliance on systemd StateDir.
+        '';
+      };
+    };
+
+
+    config = mkIf cfg.enable {
+      assertions = [
+        {
+          assertion = cfg.settings.global.server_name != null;
+          message = "services.conduit.settings.global.server_name must be set.";
+        }
+        {
+          assertion = cfg.settings.global.allow_federation -> cfg.iHaveReadTheFederationWarning;
+          message = ''
+            Conduit upstream documentation contains the following warning:
+            You should not join Matrix rooms without asking the admins first.
+            We do not know whether Conduit is safe for general use yet, so you
+            should assume there is some chance that it breaks rooms permanently
+            for all participating users. We are not aware of such a bug today,
+            but we would like to do more testing.
+            You need to acknowledge this warning by setting
+            services.conduit.iHaveReadTheFederationWarning = true
+          '';
+        }
+      ];
+
+      systemd.services.conduit = {
+        description = "Conduit Matrix Server";
+        documentation = [ "https://gitlab.com/famedly/conduit/" ];
+        wantedBy = [ "multi-user.target" ];
+        environment = lib.mkMerge ([
+          { CONDUIT_CONFIG = configFile; }
+          cfg.extraEnvironment
+        ]);
+        serviceConfig = {
+          DynamicUser = true;
```

Comment:
Can you add a `User=conduit` line here? This will only change the display user name and not actually create/use a user of that name. This is helpful for debugging.

Reply:
TIL. That's pretty cool!